src/meta-srv/src/procedure/alter_table.rs

```diff
@@ -0,0 +1,386 @@
+            ),
+            Compare::with_value(
+                table_global_key.to_string().into_bytes(),
+                CompareOp::Equal,
+                table_global_value
+                    .clone()
+                    .as_bytes()
+                    .context(error::InvalidCatalogValueSnafu)?,
+            ),
+        ]);
+
+        table_global_value.table_info = new_raw_info;
+
+        // If needs to remove a table
+        // - Removes original global value and route value. Put new global value and route value.
+        // Else
+        // - Updates original global value
+        txn = if let AlterKind::RenameTable { new_table_name } =
+            &self.data.alter_table_request.alter_kind
+        {
+            let new_table_ref = TableReference {
+                catalog: table_ref.catalog,
+                schema: table_ref.schema,
+                table: new_table_name,
+            };
+
+            let (new_table_global_key, new_table_route_key) =
+                build_table_metadata_key(new_table_ref, table_id);
+
+            txn.and_then(vec![
+                TxnOp::Delete(table_global_key.to_string().into_bytes()),
+                TxnOp::Delete(table_route_key.to_string().into_bytes()),
+                TxnOp::Put(
+                    new_table_global_key.to_string().into_bytes(),
+                    table_global_value
+                        .clone()
+                        .as_bytes()
+                        .context(error::InvalidCatalogValueSnafu)?,
+                ),
+                TxnOp::Put(
+                    new_table_route_key.to_string().into_bytes(),
+                    table_route_value.into(),
+                ),
+            ])
+        } else {
+            txn.and_then(vec![TxnOp::Put(
+                table_global_key.to_string().into_bytes(),
+                table_global_value
+                    .clone()
+                    .as_bytes()
+                    .context(error::InvalidCatalogValueSnafu)?,
+            )])
+        };
+
+        let resp = self.context.kv_store.txn(txn).await?;
+
+        ensure!(
+            resp.succeeded,
+            error::TxnSnafu {
+                msg: "table metadata changed"
+            }
+        );
+
+        self.data.state = AlterTableState::Broadcast;
+
+        Ok(Status::executing(true))
+    }
+
+    /// Broadcasts the invalidating table cache instructions.
+    async fn on_broadcast(&mut self) -> Result<Status> {
+        let table_name = self.data.table_name();
+
+        let table_ident = TableIdent {
+            catalog: table_name.catalog_name,
+            schema: table_name.schema_name,
+            table: table_name.table_name,
+            table_id: self.data.table_info.ident.table_id,
+            engine: self.data.table_info.meta.engine.to_string(),
+        };
+        let instruction = Instruction::InvalidateTableCache(table_ident);
+
+        let msg = &MailboxMessage::json_message(
+            "Invalidate Table Cache",
```

Comment:
```suggestion
            "Invalidate table cache by alter table procedure",
```
because it will be logged